json/src/main/scala/spray/json/JsReader.scala

```diff
@@ -59,8 +59,8 @@ object JsReader {
   }
 
   implicit def option[A: JsReader]: JsReader[Option[A]] = {
-    case JsNull => None
-    case a      => Some(JsReader[A].read(a))
+    case JsAbsent | JsNull => None
```

Comment:
Could you also modify the JSON writer to distinguish between an absent value and a `null`?

Reply:
I'm not sure how to do that.  `JsAbsent` is not writable.